testutil/state/suite.go

```diff
@@ -0,0 +1,104 @@
+package state
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/ovrclk/akash/testutil"
+	dkeeper "github.com/ovrclk/akash/x/deployment/keeper"
+	dtypes "github.com/ovrclk/akash/x/deployment/types"
+	"github.com/ovrclk/akash/x/market/keeper"
+	"github.com/ovrclk/akash/x/market/types"
+	pkeeper "github.com/ovrclk/akash/x/provider/keeper"
+	ptypes "github.com/ovrclk/akash/x/provider/types"
+)
+
+// TestSuite encapsulates a functional Akash nodes data stores for
+// ephemeral testing.
+type TestSuite struct {
+	t       testing.TB
+	ms      sdk.CommitMultiStore
+	ctx     sdk.Context
+	mkeeper keeper.Keeper
+	dkeeper dkeeper.Keeper
+	pkeeper pkeeper.Keeper
+	bkeeper bank.Keeper
+}
+
+// T returns the testing.T struct of the suite.
+func (ts *TestSuite) T() *testing.T {
```

Comment:
can you just return `testing.TB`?

Reply:
I agree that'd be preferable for consistency, however this is necessary because `assert` package only accepts `testing.T` :disappointed:

Reply:
which assert package?  `stretchr/testify/{assert,require}` both work with `testing.TB`.  `git grep require -- testutil`